src/main/java/cambio/simulator/entities/networking/IntermediateDependencyDescription.java

```diff
@@ -0,0 +1,104 @@
+
+    @Expose
+    private DependencyDescription[] dependencies = new DependencyDescription[0];
+
+    /**
+     * Creates a new intermediate dependency. You should use {@link #getChildren()} to modify the
+     * children.
+     * 
+     * @param model the MiSim model
+     * @param probability probability with which the dependency - and its descendants - are
+     *        executed.
+     * @param alternativeProbability probability at which the dependency - and its descendants - is
+     *        selected from a group of dependencies for execution when inside of an
+     *        {@link AlternativeDependencyDescription}
+     */
+    public IntermediateDependencyDescription(Model model, double probability,
+            double alternativeProbability) {
+        this(model, probability, alternativeProbability, new DependencyDescription[0]);
+    }
+
+    /**
+     * Creates a new intermediate dependency.
+     * 
+     * @param model the MiSim model
+     * @param probability probability with which the dependency - and its descendants - are
+     *        executed.
+     * @param alternativeProbability probability at which the dependency - and its descendants - is
+     *        selected from a group of dependencies for execution when inside of an
+     *        {@link AlternativeDependencyDescription}
+     * @param dependencies the children of this intermediate dependency. There must be at least one.
+     */
+    public IntermediateDependencyDescription(Model model, double probability,
+            double alternativeProbability, DependencyDescription[] dependencies) {
+        super(model, probability, alternativeProbability);
+        this.dependencies = dependencies;
+    }
+
+    @Override
+    public List<DependencyDescription> getChildren() {
+        return Arrays.asList(dependencies);
+    }
+
+    @Override
+    public void applyExtraDelay(final NumericalDist<Double> dist) {
+        for (DependencyDescription dependencyDescription : dependencies) {
+            dependencyDescription.applyExtraDelay(dist);
+        }
+    }
+
+    @Override
+    public void applyExtraDelay(final NumericalDist<Double> dist, final Operation operationTrg) {
+        if (operationTrg == null) {
+            throw new IllegalStateException("Target operation must not be null");
+        }
+        for (DependencyDescription dependencyDescription : dependencies) {
+            dependencyDescription.applyExtraDelay(dist, operationTrg);
+        }
+    }
+
+    @Override
+    public Set<Operation> getAllTargetOperations() {
+        return Stream.of(dependencies)
+                .flatMap(depdendency -> depdendency.getAllTargetOperations().stream())
+                .collect(Collectors.toSet());
+    }
+
+    @Override
+    public List<SimpleDependencyDescription> getLeafDescendants() {
+        return Stream.of(dependencies)
+                .flatMap(depdendency -> depdendency.getLeafDescendants().stream()).toList();
```

Comment:
Das ist scheinbar eine Java 16 API MiSim läuft noch auf Java 8.
```suggestion
                .flatMap(depdendency -> depdendency.getLeafDescendants().stream()).collect(Collectors.toList());
```